public/app/features/manage-dashboards/state/RecentlyDeletedActions.tsx

```diff
@@ -0,0 +1,58 @@
+import React from 'react';
+
+import { Button } from '@grafana/ui';
+
+import appEvents from '../../../core/app_events';
+import { Trans } from '../../../core/internationalization';
+import { useDispatch } from '../../../types';
+import { ShowModalReactEvent } from '../../../types/events';
+import { useRestoreDashboardMutation } from '../../browse-dashboards/api/browseDashboardsAPI';
+import { setAllSelection, useActionSelectionState } from '../../browse-dashboards/state';
+import { RestoreModal } from '../components/RestoreModal';
+import { useRecentlyDeletedStateManager } from '../utils/useRecentlyDeletedStateManager';
+
+export function RecentlyDeletedActions() {
+  const dispatch = useDispatch();
+  const selectedItems = useActionSelectionState();
+  const [_, stateManager] = useRecentlyDeletedStateManager();
+  const isSearching = stateManager.hasSearchFilters();
+
+  const [restoreDashboard, { isLoading: isRestoreLoading }] = useRestoreDashboardMutation();
+
+  const onActionComplete = () => {
+    dispatch(setAllSelection({ isSelected: false, folderUID: undefined }));
+
+    if (isSearching) {
+      stateManager.doSearchWithDebounce();
+    }
```

Comment:
I believe this should always be true - we don't need to do the `isSearching` check here because it's not BrowseDashboards. We're always using the Search/TrashStateManager.